src/auslib/web/public/client.py

```diff
@@ -159,81 +163,88 @@ def get_update_blob(transaction, **url):
+                    response_blobs.append({"product_query": product_query, "response_release": response_release, "response_update_type": response_update_type})
+        elif response_blob_names:
+            with statsd.timer("client.process_response_blobs"):
+                for blob_name in response_blob_names:
+                    # if we have a SuperBlob of systemaddons, we process the response products and
+                    # concatenate their inner XMLs
+                    product_query = query.copy()
+                    release_row = releases.get_release(blob_name, transaction, include_sc=False)
+                    response_release = None
+                    if release_row:
+                        product_query["product"] = releases.get_product(blob_name, transaction)
+                        response_release = createBlob(release_row["blob"])
+                    # TODO: remove me when old releases table dies
+                    else:
+                        product = dbo.releases.getReleases(name=blob_name, limit=1, transaction=transaction)[0]["product"]
+                        product_query["product"] = product
+                        response_release = dbo.releases.getReleaseBlob(name=blob_name, transaction=transaction)
+                    if not response_release:
+                        LOG.warning("No release found with name: %s", blob_name)
+                        continue
+
+                    response_blobs.append({"product_query": product_query, "response_release": response_release, "response_update_type": update_type})
         else:
             response_blobs.append({"product_query": query, "response_release": release, "response_update_type": update_type})
             # Bug 1517743 - we want a cheap test because this will be run on each request
             if release["name"] == "Firefox-mozilla-central-nightly-latest" and query["buildID"] in ("20190103220533", "20190104093221"):
                 squash_response = True
                 LOG.debug("Busted nightly detected, will squash xml response")
 
-        # getHeaderXML() returns outermost header for an update which
-        # is same for all release type
-        xml = release.getHeaderXML()
-        # we assume that all blobs will have similar ones. We might want to
-        # verify that all of them are indeed the same in the future.
-
-        # Appending Header
-        # In case of superblob Extracting Header form parent release
-        xml.append(release.getInnerHeaderXML(query, update_type, app.config["ALLOWLISTED_DOMAINS"], app.config["SPECIAL_FORCE_HOSTS"]))
-        for response_blob in response_blobs:
-            xml.extend(
-                response_blob["response_release"].getInnerXML(
-                    response_blob["product_query"], response_blob["response_update_type"], app.config["ALLOWLISTED_DOMAINS"], app.config["SPECIAL_FORCE_HOSTS"]
+    with statsd.timer("client.make_response"):
+        if release:
```

Comment:
That broke my brain a little.

We're going from:

```py
if release:
    # block a
    # block b
else:
    # ...
```

To:

```py
if release:
    # block a
with statsd(...):
    if release:
        # block b
    else:
        # ...
```

Which looks correct to me despite it being very confusing to follow

Reply:
At least the diff without whitespaces shows that something changed there, and then I emptied the condition branches to make sense of it. I don't know if there's a better way of writing this 🤷

Reply:
I've refactored this a bit in a separate commit, which should improve the readability (both of the diff and the code in general). I'll push it shortly.